docs/apm.asciidoc

```diff
@@ -0,0 +1,234 @@
++
+[source,yaml]
+----
+apiVersion: apm.k8s.elastic.co/v1alpha1
+kind: ApmServer
+metadata:
+  name: apm-server-quickstart
+  namespace: default
+spec:
+  version: "7.2.0"
+  nodeCount: 1
+  secureSettings:
+    secretName: apm-secret-settings
+  config:
+    output:
+      elasticsearch:
+        password: "${ES_PASSWORD}"
+----
+
+[float]
+[id="{p}-apm-existing-es"]
+==== Reference an existing Elasticsearch cluster
+
+Now that we know how to use the APM keystore and customize the server configuration, let's see how to manually configure a secured connection to an existing Elasticsearch cluster.
+
+. Create a secret with the Elasticsearch CA
++
+First we need to store the certificate authority of the Elasticsearch cluster in a secret:
++
+[source,yaml]
+----
+kubectl create secret generic es-ca --from-file=tls.crt=elasticsearch-ca.crt
+----
++
+Note: the file `elasticsearch-ca.crt` must contain the CA certificate of the Elasticsearch cluster you want to use with the APM server.
+
+. You can then mount this secret using the pod template, and reference the file in the `config` of the APM Server.
++
+Here is a complete example with a password stored in the Keystore as described in the previous section:
++
+[source,yaml]
+----
+apiVersion: apm.k8s.elastic.co/v1alpha1
+kind: ApmServer
+metadata:
+  name: apm-server-quickstart
+  namespace: default
+spec:
+  version: "7.2.0"
+  nodeCount: 1
+  secureSettings:
+    secretName: apm-secret-settings
+  config:
+    output:
+      elasticsearch:
+        hosts: ["my-own-elasticsearch-cluster:9200"]
+        username: elastic
+        password: "${ES_PASSWORD}"
+        protocol: "https"
+        ssl.certificate_authorities: ["/usr/share/apm-server/config/elasticsearch-ca/tls.crt"]
+  podTemplate:
+    spec:
+      containers:
+      - name: apm-server
+        volumeMounts:
+        - mountPath: /usr/share/apm-server/config/elasticsearch-ca
+          name: elasticsearch-ca
+          readOnly: true
+      volumes:
+      - name: elasticsearch-ca
+        secret:
+          defaultMode: 420
+          optional: false
+          secretName: es-ca # This is the secret that holds the Elasticsearch CA cert
+----
+
+[float]
+[id="{p}-apm-tls"]
+==== TLS Certificates
+
+By default the operator manages a private CA and generates a self-signed certificate used to secure the communication between APM agents and the server.
+
+This behavior and the relevant configuration is identical to what is done for Elasticsearch and Kibana. You can refer to link:k8s-accessing-elastic-services.html#k8s-setting-up-your-own-certificate[this documentation] to understand how to use your own certificate to configure the TLS endpoint of the APM Server.
+
+You can refer to the https://www.elastic.co/guide/en/apm/agent/index.html[APM agents documentation] for further details about how to configure them to work with custom certificates.
+
```

Comment:
For more details on how to configure them to work with custom certificates, see the https://www.elastic.co/guide/en/apm/agent/index.html[APM agents documentation].